blackduck-artifactory-common/src/test/kotlin/com/synopsys/integration/blackduck/artifactory/modules/cancel/PolicyCancelDeciderTest.java

```diff
@@ -0,0 +1,85 @@
+            Arguments.of(false, true, true, true, policyRepos, onlyMajorSeverityType, "TRIVIAL", "IN_VIOLATION", null), // Non-matching severity types
+            Arguments.of(false, true, true, true, policyRepos, onlyMajorSeverityType, null, "NOT_IN_VIOLATION", null), // Not in violation of policy
+            Arguments.of(false, true, true, true, policyRepos, onlyMajorSeverityType, "MAJOR", "IN_VIOLATION_OVERRIDDEN", null), // In violation of policy, but overridden in BlackDuck
+            Arguments.of(false, true, true, true, policyRepos, onlyMajorSeverityType, "TRIVIAL", "IN_VIOLATION_OVERRIDDEN", null), // In violation of non-matching policy, also overridden in BlackDuck
+
+            // Should cancel download for inspected artifact
+            Arguments.of(true, true, false, true, policyRepos, onlyMajorSeverityType, "CRITICAL,MAJOR", null, "IN_VIOLATION"), // OVERALL_POLICY_STATUS shouldn't matter
+            Arguments.of(true, true, false, false, policyRepos, trivialMajorSeverityTypes, "CRITICAL,MAJOR", null, "IN_VIOLATION"), // IN_VIOLATION with matching severity
+            Arguments.of(true, true, false, false, policyRepos, trivialMajorCriticalSeverityTypes, "CRITICAL,MAJOR", null, "IN_VIOLATION"), // IN_VIOLATION with multiple matching severity types
+            // Should NOT cancel download of inspected artifact
+            Arguments.of(false, false, false, false, Collections.emptyList(), Collections.emptyList(), null, null, null), // Blocking disabled
+            Arguments.of(false, true, false, false, Collections.singletonList("non-policy-repo"), Collections.emptyList(), null, null, null), // Repo not configured for policy
+            Arguments.of(false, true, false, false, policyRepos, onlyMajorSeverityType, "TRIVIAL", null, "IN_VIOLATION"), // Non-matching severity types
+            Arguments.of(false, true, false, false, policyRepos, onlyMajorSeverityType, null, null, "NOT_IN_VIOLATION"), // Not in violation of policy
+            Arguments.of(false, true, false, false, policyRepos, onlyMajorSeverityType, "MAJOR", null, "IN_VIOLATION_OVERRIDDEN"), // In violation of policy, but overridden in BlackDuck
+            Arguments.of(false, true, false, false, policyRepos, onlyMajorSeverityType, "TRIVIAL", null, "IN_VIOLATION_OVERRIDDEN") // In violation of non-matching policy, also overridden in BlackDuck
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideConfigForCancelDecision")
+    void testDecision(
+        boolean shouldCancel,
+        boolean policyBlockEnabled,
+        boolean hasScanTimeProperty,
+        boolean hasOverallPolicyStatus,
+        List<String> policyRepos,
+        List<PolicyRuleSeverityType> policyRuleSeverityTypes,
+        @Nullable String policySeverityTypesPropertyValue,
+        @Nullable String overallPolicyStatusPropertyValue,
+        @Nullable String policyStatusPropertyValue
+    ) {
```

Comment:
There are a lot of arguments here, which means lots of room for mistake in refactoring (especially with the four booleans at the start and three nullable strings at the end)

This is likely a sign that the public API of PolicyCancelDecider makes it hard to test. Something to consider for the future.

Reply:
Noted. I could probably break up the various checks into their own public methods for testing, but we have generally frowned upon that.